File: backend-laravel/app/Models/Command.php
<?php

namespace App\Models;

use Illuminate\Database\Eloquent\Concerns\HasUlids;
use Illuminate\Database\Eloquent\Factories\HasFactory;
use Illuminate\Database\Eloquent\Model;

class Command extends Model
{
    use HasFactory, HasUlids;

    public $incrementing = false;

    protected $keyType = 'string';

    protected $fillable = [
        'client_message_id',
        'device_id',
        'method',
        'params',
        'sensitive',
        'state',
        'status',
        'reason',
        'trace_id',
        'queued_at',
        'dispatched_at',
        'execution_state',
        'result',
        'error_code',
        'error_message',
        'completed_at',
        'server_seq',
        'envelope',
        'envelope_sig',
        'request_sig',
    ];

    protected $casts = [
        'params' => 'array',
        'sensitive' => 'boolean',
        'queued_at' => 'datetime',
        'dispatched_at' => 'datetime',
        'completed_at' => 'datetime',
        'result' => 'array',
        'error_code' => 'integer',
        'server_seq' => 'integer',
        'envelope' => 'array',
    ];
}

File: mobile-app/lib/models/command.dart
class CommandState {
  const CommandState({
    required this.commandId,
    required this.deviceId,
    required this.method,
    required this.state,
    required this.queuedAt,
    this.completedAt,
    this.resultStatus,
    this.errorCode,
    this.errorMessage,
    this.serverSeq,
    this.requestSig,
    this.envelopeSig,
  });

  final String? commandId;
  final String? deviceId;
  final String? method;
  final String? state;
  final String? queuedAt;
  final String? completedAt;
  final String? resultStatus;
  final int? errorCode;
  final String? errorMessage;
  final int? serverSeq;
  final String? requestSig;
  final String? envelopeSig;

  factory CommandState.fromJson(Map<String, dynamic> json) {
    final audit = json['audit'] as Map<String, dynamic>?;
    return CommandState(
      commandId: json['command_id'] as String?,
      deviceId: json['device_id'] as String?,
      method: json['method'] as String?,
      state: json['state'] as String?,
      queuedAt: json['queued_at'] as String?,
      completedAt: json['completed_at'] as String?,
      resultStatus:
          (json['result'] as Map<String, dynamic>?)?['status'] as String? ??
              json['result_status'] as String?,
      errorCode: json['error_code'] as int?,
      errorMessage: json['error_message'] as String?,
      serverSeq: (audit?['server_seq'] as num?)?.toInt(),
      requestSig: audit?['request_sig'] as String?,
      envelopeSig: audit?['envelope_sig'] as String?,
    );
  }
}
